### PR TITLE
Add metrics for pool cache

### PR DESCRIPTION
**Pull request by vkgnosis**
### Test Plan
manually tested that counters show up and increase

### CODE REVIEW

**Review by nlordell — APPROVED** [review]

File: orderbook/src/metrics.rs [hunk 1]
```diff
@@ -14,6 +14,8 @@ pub struct Metrics {
     db_table_row_count: IntGaugeVec,
     /// Outgoing RPC request metrics
     rpc_requests: HistogramVec,
+    pool_cache_hits: IntCounter,
```
Comment (fleupold) [inline orderbook/src/metrics.rs:17]: Nanonit: could also use one metric with two labels to keep this struct a bit smaller (feel free to ignore)

File: orderbook/src/main.rs [hunk 2]
```diff
@@ -221,6 +221,7 @@ async fn main() {
             args.shared.pool_cache_maximum_recent_block_age,
             Box::new(pool_aggregator),
             current_block_stream.clone(),
+            metrics.clone(),
```
Comment (fleupold) [inline orderbook/src/main.rs:224]: We should probably start using the cache in the driver as well, shouldn't we?

**Review by fleupold — APPROVED** [review]

File: orderbook/src/main.rs (hunk 2 above)
Comment (fleupold) [reply orderbook/src/main.rs:224]: #687 already done 🙈

File: orderbook/src/metrics.rs (hunk 1 above)
Comment (vkgnosis) [reply orderbook/src/metrics.rs:17]: Thought about that but I find it more natural and in line with how Prometheus docs describe new name vs new label https://prometheus.io/docs/practices/naming/ .

**Pull request merged**